qt-models/divetripmodel.cpp

```diff
@@ -752,6 +752,10 @@ void DiveTripModelTree::topLevelChanged(int idx)
 			++newIdx;
 	}
 
+	// Moving an object behind itself is a noop
+	if (newIdx == idx + 1)
```

Comment:
I think that comment is slightly misleading. Or could be read in a misleading way. If my objects are
``1 2 3 4 5``
and I move `3` to spot `4` then that isn't a no-op, it results in 
``1 2 4 3 5``
But here in this code `newIdx == idx +1` is actually the indicator that we don't need to move the entry, based on the code above it.

Reply:
Well, but in your case 3 is moved behind 4, not behind itself. Or would you read "move behind" as "move anywhere behind"?

Reply:
I'm reading the code and the comment in a rather literal way.
The new index is one bigger than the current index.
That's why I say your code is right, but your comment could be misleading.
How about
```// check if we need to move the object at all```

Reply:
Well, thinking more about this, it turns out that code, comment *and* commit message were all wrong. :)

Hopefully correct now.